File: manager-web/src/main/webapp/js/service/consultationService.js
//服务层
app.service('consultationService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../consultation/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../consultation/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../consultation/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../consultation/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../consultation/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../consultation/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../consultation/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: content-interface/src/main/java/com/service/ConsultationService.java
package com.service;

import com.shopxx.pojo.XxConsultation;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ConsultationService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<XxConsultation> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(XxConsultation consultation);
	
	
	/**
	 * 修改
	 */
	public void update(XxConsultation consultation);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public XxConsultation findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(XxConsultation consultation, int pageNum, int pageSize);
	
}
